pytket/extensions/cutensornet/backends/cutensornet_backend.py

```diff
@@ -281,18 +312,18 @@ def get_operator_expectation_value(
             )  # This needed because dagger does not work with post selection
 
         for qos, coeff in operator._dict.items():
-            expectation_value_network = ExpectationValueTensorNetwork(
+            element_network = ExpectationValueTensorNetwork(
                 bra_network, qos, ket_network
             )
             if isinstance(coeff, Expr):
                 numeric_coeff = complex(coeff.evalf())  # type: ignore
             else:
                 numeric_coeff = complex(coeff)  # type: ignore
-            expectation_term = numeric_coeff * cq.contract(
-                *expectation_value_network.cuquantum_interleaved
+            element_term = numeric_coeff * cq.contract(
+                *element_network.cuquantum_interleaved
             )
-            expectation += expectation_term
-        return expectation.real
+            element += element_term
+        return element.real
```

Comment:
It can no longer be guaranteed that the outcome is a real value, you should return a complex number.

Reply:
good point, fixed